account_cutoff_base/models/account_cutoff.py

```diff
@@ -0,0 +1,455 @@
+# Copyright 2013-2016 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountCutoff(models.Model):
+    _name = "account.cutoff"
+    _rec_name = "cutoff_date"
+    _order = "cutoff_date desc"
+    _inherit = ["mail.thread"]
+    _description = "Account Cut-off"
+
+    @api.depends("line_ids", "line_ids.cutoff_amount")
+    def _compute_total_cutoff(self):
+        for cutoff in self:
+            tamount = 0.0
+            for line in cutoff.line_ids:
+                tamount += line.cutoff_amount
+            cutoff.total_cutoff_amount = tamount
+
+    @property
+    def cutoff_type_label_map(self):
+        return {
+            "accrued_expense": _("Accrued Expense"),
+            "accrued_revenue": _("Accrued Revenue"),
+            "prepaid_revenue": _("Prepaid Revenue"),
+            "prepaid_expense": _("Prepaid Expense"),
+        }
+
+    @api.model
+    def _default_move_label(self):
+        cutoff_type = self.env.context.get("cutoff_type")
+        label = self.cutoff_type_label_map.get(cutoff_type, "")
+        return label
+
+    def _selection_cutoff_type(self):
+        # generate cutoff types from mapping
+        return list(self.cutoff_type_label_map.items())
+
+    @api.model
+    def _default_cutoff_account_id(self):
+        """Default account muast always be None"""
+        return None
+
+    @api.model
+    def _default_cutoff_journal_id(self):
+        return self.env.user.company_id.default_cutoff_journal_id
+
+    cutoff_date = fields.Date(
+        string="Cut-off Date",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        copy=False,
+        track_visibility="onchange",
```

Comment:
```suggestion
        tracking=True,
```
api of mail tred changed